.github/workflows/provenance.yml

```diff
@@ -0,0 +1,152 @@
+# Copyright The GOSST team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SLSA provenance generator
+
+permissions:
+  contents: read
+
+env:
+  # This repo.
+  GENERATOR_REPOSITORY: slsa-framework/slsa-github-generator
+  GENERATOR_REF: main
+
+###################################################################
+#                                                                 #
+#            Input and output argument definitions                #
+#                                                                 #
+###################################################################
+on:
+  workflow_call:
+    inputs:
+      artifacts:
+        description: "A comma delimited list of artifacts of the form NAME@SHA256"
+        required: true
+        type: string
+    outputs:
+      attestation-name:
+        description: "The artifact name of the signed provenance"
+        value: ${{ jobs.sign-prov.outputs.prov-name }}
+
+jobs:
+  ###################################################################
+  #                                                                 #
+  #                       Build the generator                       #
+  #                                                                 #
+  ###################################################################
+  generator:
```

Comment:
nit: Could you explain in a comment why we have two separate jobs, instead of just the build + run in a single job? The latter would avoid the need for upload/download artifact and the hash check. (I ask because I could easily see someone refactoring to merge into a single one, so if there's a reason to keep them separate, that should be documented.)

Reply:
For the Go builder, we needed to separate the "build of the builder" from the "build of the project". In provenance-only workflows, this does not seem necessary and could be removed.

Reply:
the slsa-framework is under the OpenSSF. Here's a discussion https://github.com/ossf/scorecard/issues/1534 on handling dates from folks at Linux Foundation, k8 and OpenSSF.

Note that I'm not against dates, but if they aren't truly needed I don't mind avoiding them.

Reply:
How about `SLSA Authors` in the copyright?
@ianlewis feel free to keep the date in the copyright, we can re-visit this later. No need to block this PR on this.

Reply:
SGTM

Reply:
Added license headers and merged the jobs into one.